common/presenters/framework-step-to-summary.js

```diff
@@ -1,15 +1,30 @@
-const { filter, flatten } = require('lodash')
+const { filter, find, flatten, isArray, isObject } = require('lodash')
 
 const frameworksHelpers = require('../helpers/frameworks')
 
 const frameworkFieldToSummaryListRow = require('./framework-field-summary-list-row')
 
 function frameworkStepToSummary(allFields, responses, baseUrl) {
-  return ([key, step]) => {
+  return ([key, step], i, allSteps = []) => {
+    const unmetStepConditions = allSteps
+      .map(step => step[1].next)
+      .filter(next => isArray(next))
+      .filter(next => {
+        const conditions = next
+          .filter(condition => isObject(condition))
+          .filter(condition => condition.next === step.slug)
+          .filter(item => {
+            const response = find(responses, ['question.key', item.field]) || {}
+            return item.value !== response.value
+          })
+
+        return conditions.length
+      })
```

Comment:
I think breaking this up could help a bit with understanding how it works.

1. `const allNext = allSteps.map(step => step[1].next).filter(next => isArray(next))`. It is the same for all steps.
1. `isArray(next) === isObect(condition)` since `isObject([]) === true`. This can be removed.
1. You're interested only in the set that is relevant to the current step. `const relevantNext = allNext.filter(/* */)`
1. Find the answers

You could have a function that retrieves all previousSteps

Reply:
I'd love if you had time to post a snippet. I toyed with this a lot to get it to return what it actually needs to.

I think 2. isn't possible because they are checks on separate objects. When is check that `next` is in fact an array, because in most cases it's just a string. And then `isObject` is checking that the condition is an object and therefore is dependent on a responses value.

Here's an example of what nexts can look like from the test file:
```
[
  '/step-one',
  {
    next: [
      {
        field: 'conditionalField',
        value: 'Yes',
        next: 'step-two',
      },
      'step-three',
    ],
    slug: 'step-one',
    fields: ['fieldOne'],
  },
  ],
  [
    '/step-two',
    {
      next: 'step-three',
      slug: 'step-two',
      fields: ['fieldOne'],
    },
  ],
  [
    '/step-three',
    {
      slug: 'step-three',
      fields: ['fieldOne'],
    },
  ],
]
```

`next` supports the same logic from the [form wizard](https://github.com/UKHomeOffice/passports-form-wizard#next-steps). So the function of this is to remove steps from the display if the condition is not met.

Reply:
This snippet passes the current tests:

```js
    const allNextSteps = allSteps
      .map(([key, step]) => step.next)
      .filter(next => isArray(next))
      .reduce((acc, it) => [...it, ...acc], [])

    const unmetStepConditions = allNextSteps
      .filter(condition => condition.next === step.slug)
      .filter(item => {
        const response = find(responses, ['question.key', item.field]) || {}
        return item.value !== response.value
      })
```